agent/run.py

```diff
@@ -34,22 +35,19 @@ async def disconnect(self, websocket: WebSocket):
         del self.sender_tasks[websocket]
         del self.message_queues[websocket]
 
-    async def start_streaming(self, task, report_type, agent, agent_role_prompt, websocket):
-        report, path = await run_agent(task, report_type, agent, agent_role_prompt, websocket)
+    async def start_streaming(self, task, report_type, websocket):
+        report, path = await run_agent(task, report_type, websocket)
         return report, path
 
 
-async def run_agent(task, report_type, agent, agent_role_prompt, websocket):
-    check_config_setup()
+async def run_agent(task, report_type, websocket):
+    researcher = GPTResearcher()
```

Comment:
@gregdrizz a bit confused here but shouldn't GPTResearcher be constructed and initialized with a task and stream type? Then you can call "conduct_researcher" with a specific agent maybe, not sure if that as well should be in construction step.